modules/integration/tests/src/test/java/org/wso2/appmanager/integration/utils/APPMPublisherRestClient.java

```diff
@@ -0,0 +1,382 @@
+        String acsUrl = appCreateRequest.getOverviewAcsUrl();
+
+        String requestBody = "{\"provider\":\"" + provider +
+                "\",\"logout_url\":\"" + logOutUrl +
+                "\",\"app_acsURL\":\"" + acsUrl +
+                "\",\"claims\":[\"" + claims + "\"],\"app_name\":\"" + appName +
+                "\",\"app_verison\":\"" + version +
+                "\",\"app_transport\":\"" + transport +
+                "\",\"app_context\":\"" + context +
+                "\"}";
+        this.requestHeaders.put(AppmTestConstants.CONTENT_TYPE, "application/json");
+        HttpResponse response =
+                HttpUtil.doPost(new URL(
+                        backEndUrl + AppmTestConstants.PubliserRestApis.ADD_SSO_PROVIDER),
+                                requestBody, requestHeaders);
+        return response;
+    }
+
+    /**
+     * Add the roles to an application.
+     *
+     * @param roles String.
+     * @param appId String.
+     * @return roleAddedResponse HttpResponse.
+     * @throws Exception on errors.
+     */
+    private HttpResponse addRole(String roles, String appId) throws Exception {
+        String role = roles;
+        this.requestHeaders.put("Content-Type", "application/json");
+        HttpResponse response =
+                HttpUtil.doPost(new URL(backEndUrl + "/publisher/asset/webapp/id/" +
+                                                appId + "/permissions"),
+                                "[{\"role\":\"" + role +
+                                        "\",\"permissions\":[\"GET\",\"PUT\",\"DELETE\"," +
+                                        "\"AUTHORIZE\"]}]",
+                                requestHeaders);
+        if (response.getResponseCode() == 200) {
+            return response;
+        } else {
+            throw new Exception("Add role failed> " + response.getData());
+        }
+    }
+
+    /**
+     * Publish an application.
+     *
+     * @param appId String.
+     * @return appPublishedResponse String.
+     * @throws Exception on errors.
+     */
+    public HttpResponse publishWebApp(String appId) throws Exception {
+
+        //Submit the app for review
+        changeState(appId, "Submit for Review");
+        //Approve the app
+        changeState(appId, "Approve");
+        //Publish the app
+        HttpResponse response = changeState(appId, "Publish");
+        return response;
+    }
+
+    /**
+     * Change the life cycle state from current state to next state
+     *
+     * @param appId   String.
+     * @param toState String.
+     * @return changedStateResponse HttpResponse.
+     * @throws Exception on errors.
+     */
+    public HttpResponse changeState(String appId, String toState) throws Exception {
+        checkAuthentication();
+        this.requestHeaders.put(AppmTestConstants.CONTENT_TYPE, "");
+        String encodedState = toState.replaceAll(" ", "%20");
+
+        HttpResponse response =
```

Comment:
Fixed in https://github.com/wso2/product-app-manager/commit/f021f86c3c398dddeeca17e225a37e42f7f7033b